packages/synapse-react-client/src/components/SynapseTable/SynapseTable.tsx

```diff
@@ -1,889 +1,275 @@
-import { SynapseTableCell } from './SynapseTableCell/SynapseTableCell'
-import { Checkbox } from '../widgets/Checkbox'
-import { EnumFacetFilter } from '../widgets/query-filter/EnumFacetFilter'
-import { ICON_STATE } from './SynapseTableConstants'
+import { useQueryVisualizationContext } from '../QueryVisualizationWrapper/QueryVisualizationWrapper'
+import { useQueryContext } from '../QueryContext'
 import {
-  getColumnIndicesWithType,
-  getUniqueEntities,
+  isEntityViewOrDataset,
   isFileViewOrDataset,
   isSortableColumn,
 } from './SynapseTableUtils'
 import { TablePagination } from './TablePagination'
-import EntityIDColumnCopyIcon from './EntityIDColumnCopyIcon'
 import ExpandableTableDataCell from './ExpandableTableDataCell'
-import { Box, IconButton, Tooltip } from '@mui/material'
-
-type Direction = '' | 'ASC' | 'DESC'
-export const SORT_STATE: Direction[] = ['', 'DESC', 'ASC']
-export const DOWNLOAD_OPTIONS_CONTAINER_CLASS = 'SRC-download-options-container'
-const RESIZER_OPTIONS: any = {
-  resizeMode: 'overflow',
-  partialRefresh: 'true',
-  liveDrag: true,
-  headerOnly: 'true',
-}
-type Info = {
-  index: number
-  name: string
-}
-
-export type SynapseTableState = {
-  sortedColumnSelection: SortItem[]
-  columnIconSortState: number[]
-  isExportTableDownloadOpen: boolean
-  isExpanded: boolean
-  mapEntityIdToHeader: Record<string, EntityHeader>
-  mapUserIdToHeader: Record<string, UserGroupHeader | UserProfile>
-  isColumnSelectionOpen: boolean
-  isFetchingEntityHeaders: boolean
-}
+import { Box } from '@mui/material'
+import {
+  ColumnDef,
+  createColumnHelper,
+  flexRender,
+  getCoreRowModel,
+  useReactTable,
+  VisibilityState,
+} from '@tanstack/react-table'
+import { useTableSort } from './useTableSort'
+import {
+  accessColumn,
+  addToDownloadListColumn,
+  directDownloadColumn,
+  rowSelectionColumn,
+  TableDataCell,
+  TableDataColumnHeader,
+} from './SynapseTableRenderers'
+import { SynapseTableContext } from './SynapseTableContext'
 
 export type SynapseTableProps = {
-  synapseContext: SynapseContextType
-  queryContext: QueryContextType
-  queryVisualizationContext: QueryVisualizationContextType
-  title?: string
+  /** If true and entity is a view or dataset, renders a column that represents if the caller has permission to download the entity represented by the row */
   showAccessColumn?: boolean
+  /** @deprecated use showDirectDownloadColumn */
   showDownloadColumn?: boolean
+  /** If true and entity is a file view or dataset, renders a column that allows the logged-in user to directly download the file, if they have permission */
+  showDirectDownloadColumn?: boolean
+  /** If true, the add to download list column will be hidden */
+  hideAddToDownloadListColumn?: boolean
+  /** Configuration to override cell renderers with e.g. a link to a portals detail page */
   columnLinks?: LabelLinkConfig
-  hideDownload?: boolean
 }
 
-export class SynapseTable extends React.Component<
```

Comment:
IMO not worth looking at this side of the diff, since this was a complicated class component with a bunch of unused code.